atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbErrorDecoder.java

```diff
@@ -43,6 +43,13 @@ public Exception decode(String methodKey, Response response) {
         if (response503ButExceptionIsNotRetryable(response, exception)) {
             return new RetryableException(exception.getMessage(), exception, parseRetryAfter(response));
         }
+        if (response429ButExceptionIsNotRetryable(response, exception)) {
+            // We want to retry on the same node (ExceptionRetryBehaviour.RETRY_ON_SAME_NODE) every time we receive
+            // a response with status 429. Therefore, we set retryAfter to null to comply with
+            // ExceptionRetryBehaviour.getRetryBehaviourForException logic.
+            return new RetryableException(exception.getMessage(), exception, null);
```

Comment:
Okay. Note that with `FailoverFeignTarget` the way it is, the behaviour we'll be getting is that:

1. We will retry up to 3 times on the leader. Let's assume we get three 429s because threads are busy doing other things.
2. We will retry once each on each non-leader.
3. Go back to 1.

My main concern here is that we might be generating large amounts of network traffic, since these steps could potentially happen rather quickly. I think this is ok for this PR (is already large enough), but we should think about this and either fix it or decide it's safe before recommending people turn the rate limiter on in general.